File: Corrections/php/Neufplate/src/AvatarProvider/DiceBear/DiceBearAvatar.php
<?php

namespace App\AvatarProvider\DiceBear;

use App\Avatar;
use App\AvatarClientInterface;
use App\AvatarProvider\SpriteType;


class DiceBearAvatar extends Avatar
{

    private SpriteType $spriteType;

    public function __construct(SpriteType $spriteType = SpriteType::DEFAULT)
    {
        $this->spriteType = $spriteType;
    }


    protected function getClient(): AvatarClientInterface
    {
        $client = DiceBearClient::getInstance();
        $client->setSpriteType($this->spriteType);
        return $client;
    }
}

File: Corrections/java/Neufplate/src/main/java/fr/neufplate/AvatarProvider/DiceBear/DiceBearAvatar.java
package fr.neufplate.AvatarProvider.DiceBear;

import fr.neufplate.Avatar;
import fr.neufplate.AvatarClientInterface;
import fr.neufplate.AvatarProvider.SpriteType;

public class DiceBearAvatar extends Avatar {

    private final SpriteType spriteType;

    public DiceBearAvatar(SpriteType spriteType) {
        if (spriteType == null) {
            throw new IllegalArgumentException("SpriteType cannot be null");
        }
        this.spriteType = spriteType;
    }

    @Override
    protected AvatarClientInterface getClient() {
        DiceBearClient client = DiceBearClient.getInstance();
        client.setSpriteType(this.spriteType);
        return client;
    }
}
